libs/build_condaenv.sh

```diff
@@ -21,22 +21,30 @@ rqmts_file=${HPC_STACK_ROOT}/pyvenv/$rqmts
 [[ ! -f $rqmts_file ]] && ( echo "Unable to find environment file: $rqmts \nABORT!"; exit 1 )
 
 python=$(echo $HPC_PYTHON | sed 's/\//-/g')
+echo "STACK_miniconda3_version = ${STACK_miniconda3_version:-}"
 
 if $MODULES; then
   set +x
   source $MODULESHOME/init/bash
-  module load hpc-$HPC_PYTHON
+  module load hpc
+  if [[ -n "${STACK_miniconda3_version:-} " ]]; then
+    module is-loaded python  && module unload python
+    module load miniconda3/${STACK_miniconda3_version:-}
+  else
+    module load hpc-$HPC_PYTHON
+  fi
```

Comment:
This is a bit tricky. When miniconda3 is requested to be being built as a part of the hpc-stack (and is not used as as system module) , HPC-python environment needs to be  set "HPC-python" not "HPC-miniconda3". This is needed for the other modules to be built prior to miniconda3. 

Variable ${STACK_miniconda3_version} is set only when the miniconda3 is built as a part of the stack. The module miniconda3/${STACK_miniconda_version} could only be loaded when a python module (if a system-wide module exists) is unloaded.

Any ideas?

Reply:
If miniconda3 is build locally and not as a part of `hpc-stack`, the module loaded will be `hpc-miniconda3`. So there seems to be no conflict. I could only suggest to use $SUDO to create a conda environment, in case the system-wide installation of miniconda3 exist:

$SUDO conda env create --file $rqmts_file

Reply:
@aerorahul @kgerheiser -
I addressed the issue by creating both virtual environment and the module with the same name. The summary of changes on the top of the PR description  now reads:

- Create a virtual environment regional_workflow with the packages needed for the SRW, when it is requested in stack_....yaml. Use and adapted ./libs/build_condaenv.sh script. It also builds the module with the same name regional_worflow. Virtual environment could then be activated/deactivated in two ways:
a) Load miniconda3 module, and then by using "conda activate regional_workflow" , "conda deactivate"
b) Load hpc-${HPC_PYTHON} module, miniconda3 module, and then use "module load regional_workflow" , "module unload regional_workflow".

Reply:
Please see attached a screenshot from the build on Orion, showing the two ways the conda environment is activated
![conda_hpc-python_regional_workflow](https://user-images.githubusercontent.com/68030316/179307484-7e94f3c1-35ee-44d7-a2b2-77e6e0ccdfe7.png)